gui/layers.py

```diff
@@ -754,41 +750,38 @@ def layer_visible_pixbuf_datafunc(column, cell, model, it, data):
     layer = model.get_layer(it=it)
     rootstack = model._root
     visible = True
-    greyed_out = True
+    sensitive = True
     if layer:
         # Layer visibility is based on the layer's natural hidden/
         # visible flag, but the layer stack can override that.
-        visible = layer.visible
-        greyed_out = False
         if rootstack.current_layer_solo:
-            visible = (layer is rootstack.current)
-            greyed_out = True
-        elif DISTINGUISH_DESCENDENTS_OF_INVISIBLE_PARENTS:
-            path = model.get_path(it).get_indices()
-            path.pop()
-            while len(path) > 0:
-                ancestor = model.get_layer(treepath=path)
-                if not ancestor.visible:
-                    greyed_out = True
-                    break
-                path.pop()
-    # Pick icon
-    icon_name_template = "mypaint-object{vis}{sens}-symbolic"
-    icon_name = icon_name_template.format(
-        vis=("-visible" if visible else "-hidden"),
-        sens=("-insensitive" if greyed_out else ""),
+            visible = layer is rootstack.current
+            sensitive = False
+        else:
+            visible = layer.visible
+            sensitive = layer.branch_visible
+
+    icon_name = "mypaint-object-{}-symbolic".format(
+        "visible" if visible else "hidden",
     )
     cell.set_property("icon-name", icon_name)
+    cell.set_property("sensitive", sensitive)
```

Comment:
Thanks. The name hack always seemed an ungreppable mess to me, and this change doesn't seem to have any bad consequences like making the icon unclickable. At least it doesn't for Adwaita and GTK 3.20.4 :eyes: